proposals/0248-tpu-feedback.md

```diff
@@ -0,0 +1,178 @@
+---
+simd: '0224'
+title: TPU Feedback
+authors:
+  - Lijun Wang <lijun@anza.xyz>
+category: Standard
+type: Core
+status: Review
+created: 2025-02-14
+development:
+  - Anza - WIP
+  - Firedancer - Not started
+---
+
+## Summary
+
+This document defines the protocol to send feedback on TPU processing from the
+servers to the clients.
+
+## Motivation
+
+A client sends TPU transactions to the leader as QUIC uni-streams. Currently we
+only send transport level acknowledememt to the clients. I.e we only tell if a
+transaction has been received by the server. A transaction received by the
+server may not be processed due to resource contention. For example, if there
+are too many transactions with higher paying fees than this transaction, the
+transaction is not included in the block. Providing further information on the
+transaction state will help the client to make more informed decision on
+handling transaction failures and reduce the incentive to blindly resubmitting
+transactions and help reduce congestion condition in the network.
```

Comment:
> reduce the incentive to blindly resubmitting transactions and help reduce congestion condition in the network

imo, by providing feedbacks itself does not necessarily disincentivize spamming; but Server could use `TransactionState` to throttle connections with excessively high failure rate, or use `PriorityFeeInfo` to drop transactions have priority below `min`. Are these reasonable use cases for feedbacks for this SIMD?

Reply:
The client can use the information to 

1. modify the retry behavior of transactions
2. proactively not sending transactions which are seriously below the water mark
3. percolate the information and set higher fee for transactions to maximize the chance to be included.